sycl/doc/design/CompileTimeProperties.md

```diff
@@ -597,18 +603,20 @@ types listed above.
 
 Properties that are implemented using
 `[[__sycl_detail__::add_ir_annotations_member()]]`, are represented in LLVM IR
-as the fifth metadata parameter to the `@llvm.ptr.annotation` intrinsic
-function.  This parameter is a tuple of metadata values with the following
-sequence:
+as the fifth parameter to the `@llvm.ptr.annotation` intrinsic function.  This
+parameter is a pointer to a global variable with fields corresponding to the
+names and values of the properties in the following sequence:
 
 * Name of the first property
 * Value of the first property
 * Name of the second property
 * Value of the second property
 * Etc.
 
-Since metadata types are not limited to strings, there is no need to convert
-the property values to strings.
+Every field in the global variable pointed to by this parameter are string
+literals in seperate global variables. If the value of a property is either an
+empty string or a `nullptr`, the corresponding field will have a null-pointer
+value.
```

Comment:
Could we change this last sentence like:

> Property values are converted to strings in the same way as described above, except that the `nullptr` value is represented as `nullptr` in the global variable field.

This is kind of an edge case, but if the original property value in the C++ attribute is the empty string (`""`), shouldn't that be represented as an empty string in the global variable field?  In your wording, it would be represented as a `nullptr`.

Reply:
> This is kind of an edge case, but if the original property value in the C++ attribute is the empty string (""), shouldn't that be represented as an empty string in the global variable field? In your wording, it would be represented as a nullptr.

We can do that, but other for the ones that add IR attributes, an empty string means that the generated attribute won't have a value, so the existing logic simply uses the empty string as an indicator for `nullptr` as well. For consistency I prefer keeping it that way. For properties that may want an empty string in the decorations, sycl-post-link will have to recognize them anyway and can simply reintroduce the empty string at that point.

Reply:
OK, I have no concerns with that.  The sentence should still say something about the way property values are converted to strings.  Maybe:

> Property values are converted to strings in the same way as described above, except that the `nullptr` value and the empty string (`""`) is represented as `nullptr` in the global variable field.

Reply:
Sounds good! I have rephased the sentence using your suggestion.